amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
@marcoceppi I'd rather keep it simple and just have JUJU_TEST_CONSTRAINTS, with author's constraints always overriding them.